### ISSUE TITLE
Getting session token fails when the extension is added the first time before the checkout is saved

### ISSUE THREAD

**Issue by TheSecurityDev** [Shopify/ui-extensions#1081]
## Please list the package(s) involved in the issue, and include the version you are using

` "@shopify/checkout-ui-extensions-react": "^0.27.0"`

## Describe the bug

The `useSessionToken` hook returns a function that you can use to get a session token every time you make a request to your backend from the extension. But when you implement it, fetching the session token throws an error when you've added the extension in the checkout editor but haven't saved it yet.

**This can't be reproduced in the dev extensions when using the local dev server.**

## Steps to reproduce the behavior:

1. Use the `useSessionToken` hook to get a session token getter function.
1. Use that getter function to get a session token on each request, or do it right when the extension is rendered for testing.
1. Publish the extension (you can do it from the working draft on a dev store).
2. Go to the checkout editor and add the extension to the page (make sure it's not already added).
3. Observe that fetching the session token fails.
4. Save the checkout with the extension added and refresh the page.
5. Observe that it now works.

## Expected behavior

The merchant should be able to use the extension in the checkout editor without saving it first and making it live. We need the requests to start working immediately.

## Screenshots

![image](https://github.com/Shopify/ui-extensions/assets/49467229/f094b688-f928-42bf-93e1-f8865f84a3b7)

## Additional context

In the network panel we can see that it tries to fetch the session token, but gets the following response:

```json
{
    "data": {
        "checkoutExtensionSessionToken": null
    }
}

```

**Comment by dnagoda** [comment]
Thanks for the report. Adding to the backlog.

**Comment by luklapp** [comment]
@dnagoda Any update on this issue? I can still reproduce it with version `2023.10.0`

**Comment by TheSecurityDev** [comment]
Yeah, it's a pretty poor experience for the merchant as we have to check for the error and show a banner telling them to save the checkout and refresh first. But the point of previewing is to be able to test it before going live, and not all merchants might think to create a draft checkout first.
